pkg/services/hivelocity/device/device.go

```diff
@@ -328,19 +353,58 @@ func (s *Service) actionVerifyShutdown(ctx context.Context) actionResult {
 	// remove condition to reset the timer - we set the condition anyway again
 	conditions.Delete(s.scope.HivelocityCluster, infrav1.DeviceProvisioningSucceededCondition)
 
-	// TODO: make shut down API call
-
+	err = s.scope.HVClient.PowerOffDevice(ctx, deviceID)
+	if err != nil {
+		return actionError{err: fmt.Errorf("[actionDeviceProvisioned] PowerOffDevice failed: %w", err)}
+	}
 	conditions.MarkFalse(
 		s.scope.HivelocityCluster,
 		infrav1.DeviceProvisioningSucceededCondition,
 		infrav1.DeviceShutdownCalledReason,
 		clusterv1.ConditionSeverityInfo,
 		"device shut down has been triggered",
 	)
-
 	return actionContinue{delay: 30 * time.Second}
 }
 
+// Set permanent error with an appropriate label,
+// and go back to the state associate to associate with another device via actionGoBack.
+func (s *Service) setReloadingTooLongTag(ctx context.Context, deviceID int32, provisionCondition *clusterv1.Condition) actionResult {
```

Comment:
I think you could use the deviceID as function argument.. Or do you think we will re-use this function somewhere where we don't have it?

Reply:
@janiskemper what do you mean with
 
 > use the deviceID as function argument.
 
 Do you think it is better to create a function instead of a method (of Service)?

Reply:
I really misunderstood the code, sorry.. I thought you get the device object "in vain" because you don't pass the right arguments.. But you have to call GetDevice because you haven't done it before at any point.... 
Resolved!